src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy

```diff
@@ -0,0 +1,73 @@
+package org.gradle.crypto.checksum
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class IncrementalChecksumSpec extends Specification {
+
+    @Rule
+    TemporaryFolder projectDir = new TemporaryFolder()
+
+    def setup() {
+        newFile('settings.gradle')
+    }
+
+    def 'can handle changing file trees'() {
+        given:
+        newFile('build.gradle') << """
+            plugins {
+                id 'org.gradle.crypto.checksum' version '1.1.0'
+            }
+        
+            import org.gradle.crypto.checksum.Checksum
+            
+            task checksums(type: Checksum) {
+                files = fileTree("input") {
+                    include "**/*.txt"
+                }
+                outputDir = file("checksums")
+            }
+        """
+        newFolder('input')
+        newFile('input/foo.txt') << 'foo'
```

Comment:
Should we add some files deeper in the directory structure? Note that even with pre-5.0 Gradle versions, the build probably would have failed, given that non-root directories have been reported as changes before as well.

Reply:
Makes sense. Thanks.